src/GraphQL.Tests/Bugs/Bug781UnobservedTasks.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using GraphQL.SystemTextJson;
+using GraphQL.Types;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Tests.Bugs
+{
+    public class Bug781UnobservedTasks
+    {
+        private bool _unobserved;
+
+        [Theory]
+        [InlineData("{ do(throwCanceled: false) cancellation never }", true)]
+        [InlineData("{ do(throwCanceled: true) cancellation never }", false)]
+        public async Task cancel_execution_with_different_error_types(string query, bool unobserved)
+        {
+            TaskScheduler.UnobservedTaskException += TaskScheduler_UnobservedTaskException;
+            var cts = new CancellationTokenSource();
+            ISchema schema = new Bug781Schema(cts);
+            string result = null;
+
+            try
+            {
+                result = await schema.ExecuteAsync(options =>
+                {
+                    options.Query = query;
+                    options.CancellationToken = cts.Token;
+                    options.ThrowOnUnhandledException = true; // required
+                });
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+            }
+
+            GC.Collect(); // GC causes UnobservedTaskException event
+            await Task.Delay(1000); // Wait some time for GC to complete
```

Comment:
```suggestion
            GC.Collect(); // call finalizer of Task
            await Task.Delay(1000); // Wait some time for GC to complete
            GC.Collect(); // Task is collected and GC causes UnobservedTaskException event
            await Task.Delay(1000); // Wait some time for GC to complete
```

Reply:
This fixes the test, at least on my machine.  Why?  I don't know; I'd just be guessing.  Presumably you had the code working before.  Another example of the test being unreliable.